openstack/sharedfilesystems/v2/shares/util.go

```diff
@@ -0,0 +1,22 @@
+package shares
+
+import (
+	"github.com/gophercloud/gophercloud"
+)
+
+// WaitForStatus will continually poll the resource, checking for a particular
+// status. It will do this for the amount of seconds defined.
+func WaitForStatus(c *gophercloud.ServiceClient, id, status string, secs int) error {
```

Comment:
@jrperritt I decided to move this under the acceptance tests of shared file systems, what do you think? It's not optimal but generalizing requires more work which would probably warrant it's own PR.